src/components/ModuleTeacherCards/ModuleTeacherCards.tsx

```diff
@@ -0,0 +1,125 @@
+import CardSkeleton from 'components/CardSkeleton/CardSkeleton';
+
+import { ReactComponent as LockIcon } from 'routes/modules/lock.svg';
+import { ReactComponent as UnlockIcon } from 'routes/modules/unlock.svg';
+
+import { Module } from 'types/Module/Module';
+
+import { useSelector } from 'react-redux';
+
+import { RootState } from 'store/store';
+
+import style from './ModuleTeacherCards.module.css';
+
+type Props = {
+  data: Module[];
+  initialState: Module[];
+  setModuleData: Function;
+  onUpdate: Function;
+};
+
+export function ModuleTeacherCards({
+  data,
+  initialState,
+  setModuleData,
+  onUpdate,
+}: Props) {
+  const isLoading = useSelector((state: RootState) => state.loading.loadingVal);
+
+  const [editing, setEditing] = useState(false);
+
+  const hasChanges = data.reduce((result, current, index) => {
+    if (current.is_active !== initialState[index].is_active) result = true;
+    return result;
+  }, false);
+
+  function onCancel() {
+    setEditing(false);
+    setModuleData(initialState);
+  }
+
+  const onClickHandler = async () => {
+    if (editing) {
+      onUpdate();
+      setEditing(false);
+    } else {
+      setEditing(true);
+    }
+  };
+
+  function changeModuleState(index: number) {
+    if (editing) {
+      setModuleData(
+        data.map((module, i) => {
+          if (i === index) return { ...module, is_active: !module.is_active };
+          return module;
+        })
+      );
+    }
+  }
+
+  const cardStyles = (i: number) => {
+    if (data[i].is_active) return style.card;
+    return `${style.card} ${style.locked}`;
+  };
+
+  if (isLoading) {
+    return (
+      <div className={style['modules-container']}>
+        <div className={style.modules}>
+          <CardSkeleton items={8} />
+        </div>
+      </div>
+    );
+  }
+
+  return (
+    <div className={style['modules-container']}>
+      <div className={style['button-container']}>
+        <Button
+          location="modules"
+          text={editing ? 'Guardar cambios' : 'Editar visibilidad'}
+          isDisable={editing ? !hasChanges : false}
+          onClickHandler={() => onClickHandler()}
+        />
+        {editing && (
+          <Button
+            location="modules"
+            className={style.cancel}
+            text="Cancelar"
+            isDisable={false}
+            onClickHandler={() => onCancel()}
+          />
+        )}
+      </div>
+      <div className={style.modules}>
+        {data.map((module, i) => (
+          <Card
+            key={module.title}
+            className={cardStyles(i)}
+          >
+            <h2 className={style.title}>
+              {i + 1}. {module.title}
+            </h2>
+            <div className={style['icon-container']}>
+              {data[i].is_active ? (
+                <UnlockIcon
+                  className={editing ? style['clickable-icon'] : style.icon}
+                  onClick={() => changeModuleState(i)}
+                />
+              ) : (
```

Comment:
According to our findings in the SUS assessment, the teachers believe that the card should be clickable, not only the lock icon. We should change this.

Reply:
I am not so sure if the card should be clickable. I understood that the problem was that the user perceived the card as clickable because of the scale that happened on hover, so I removed it from the card.
I checked with the SUS tester and he liked this change, but we could discuss this further